### PR TITLE
feat(web): implement multiple users per workspace

### PR DESCRIPTION
**Pull request by paulocsanz**
- Invite users by email
- We do not validate emails for password logins
- Hard reload page if workspace changes to avoid reactivity issues

image
image

### CODE REVIEW

File: lib/dal/src/user.rs [hunk 1]
```diff
@@ -137,6 +155,41 @@ impl User {
...
+        // Race condition here, if the user signs-up while we are inviting them the invite will be
+        // lost forever
+        ctx.txns()
+            .await?
+            .pg()
+            .execute(
+                "SELECT user_invite_to_workspace_v1($1, $2)",
+                &[&email, &workspace_pk],
+            )
+            .await?;
```
Comment (nickgerace) [inline lib/dal/src/user.rs:178]: If the DB isn't where the hang occurs, maybe wrap this in a `tokio::time::timeout`?

File: lib/sdf-server/src/server/extract.rs [hunk 2]
```diff
@@ -132,9 +132,17 @@ impl FromRequestParts<AppState> for Authorization {
...
+            .get("WorkspacePk")
+            .and_then(|w| w.to_str().ok())
+            .and_then(|id| WorkspacePk::from_str(id).ok());
+        let claim = UserClaim::from_bearer_token(
+            workspace_pk_header_value,
+            jwt_public_signing_key,
+            authorization,
+        )
+        .await
+        .map_err(|_| unauthorized_error())?;
```
Comment (nickgerace) [inline lib/sdf-server/src/server/extract.rs:145]: Beautiful

File: lib/dal/src/user.rs (hunk 1 above)
Comment (paulocsanz) [reply lib/dal/src/user.rs:178]: I'm not sure I follow

**Comment by paulocsanz** [comment]
bors merge

**Comment by paulocsanz** [comment]
bors merge

**Pull request merged**